### PR TITLE
Added hints about receiving feedback

### PR DESCRIPTION
**Pull request by lsmith77**
relates to https://github.com/symfony/diversity/issues/23

Changes made:
* define and use the term PR rather than patch
* added hint to first check existing issues/PR
* added hints and relevant links about receiving feedback
* several other minor changes 

todo:
* [x] rename file to PR.rst and add entry to the redirect map

### CODE REVIEW

File: contributing/code/patches.rst [hunk 1]
```diff
@@ -90,20 +99,25 @@ Check that the current Tests Pass
 Now that Symfony is installed, check that all unit tests pass for your
 environment as explained in the dedicated :doc:`document <tests>`.
 
-Step 2: Work on your Patch
---------------------------
+.. tip::
+
+    If tests are failing, check on `Travis-CI`_ if the tests are failing
+    there as well. Normally they should be green but if they also fail
+    there you don't have to alarmed if you also fail locally.
```
Comment (OskarStark) [inline contributing/code/patches.rst]: This sentence sounds weird

**Review by OskarStark — COMMENTED** [review]
You should rename the file, too and add it to the redirection map.
But maybe doing this after merging this, otherwise the diff isn’t readable anymore

File: contributing/code/patches.rst [hunk 2]
```diff
@@ -374,3 +405,7 @@ before merging.
 .. _`fabbot`: https://fabbot.io
 .. _`PSR-1`: https://www.php-fig.org/psr/psr-1/
 .. _`PSR-2`: https://www.php-fig.org/psr/psr-2/
+.. _``searching on GitHub: https://github.com/symfony/symfony/issues?q=+is%3Aopen+
```
Comment (wouterj) [inline contributing/code/patches.rst]: Typo here with the backticks

File: contributing/code/patches.rst [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
+enhancements to Symfony.
 
-Step 1: Setup your Environment
+Step 1: Check existing issues and PRs
+-------------------------------------
+
+Before working on a change, check to see if someone else also raised the topic
+or maybe even started working on a PR by `searching on GitHub`_.
+
+If you are still unsure or if you have any questions during this entire process,
```
Comment (wouterj) [inline contributing/code/patches.rst]: I would remove "still" here. At the moment, it sort of looks like you have to ask on slack if you didn't find a related issue?

File: contributing/code/patches.rst (hunk 1 above)
Comment (wouterj) [reply contributing/code/patches.rst]: "[...] to be alarmed."

Comment (lsmith77) [reply contributing/code/patches.rst]: tweaked it slightly .. better now?

**Comment by lsmith77** [comment]
added a todo item for the rename/redirect

**Review by wouterj — APPROVED** [review]
Great changes in here! I've left some very minor comments

**Comment by wouterj** [comment]
Oh and one more thing: We provide "backwards compatiblity" for section titles. That means that if you change a title "Start working on your patch" to "Start working on your PR", we have to add another anchor for the old title (`_start-working-on-your-patch:` in this example) above it

File: contributing/code/patches.rst [hunk 4]
```diff
@@ -90,20 +99,25 @@ Check that the current Tests Pass
...
+-----------------------
 
 The License
 ~~~~~~~~~~~
 
-Before you start, you must know that all the patches you are going to submit
-must be released under the *MIT license*, unless explicitly specified in your
-commits.
+Before you start, you must know that all the code you are going to submit
+must be released under the *MIT license*.
```
Comment (dbrumann) [inline contributing/code/patches.rst]: not sure, but I think the double "must" makes it sound a bit off, maybe use will instead?

```suggestion
will be released under the *MIT license*.
```

File: contributing/code/patches.rst [hunk 5]
```diff
@@ -167,8 +181,8 @@ uses, and replaces them by symbolic links to the ones in the Git repository.
 Before running the ``link`` command, be sure that the dependencies of the project you
 want to debug are installed by running ``composer install`` inside it.
 
-Work on your Patch
-~~~~~~~~~~~~~~~~~~
+Work on your PR
```
Comment (linaori) [inline contributing/code/patches.rst]: In the text it makes sense to refer to PR, in the title I think it looks better to refer to a "Pull Request" though. It doesn't actually bloat anything and works better when creating a list based on the page titles, as PR is not explained in that list. I think the same goes for all other titles as well

**Comment by lsmith77** [comment]
@wouterj did I handle the BC for the section titles correctly now?

File: contributing/code/patches.rst [hunk 6]
```diff
@@ -339,11 +366,29 @@ commit message).
...
-~~~~~~~~~~~~~~~~~
+Step 5: Receiving Feedback
+--------------------------
+
+We ask all contributors to follow some
+:doc:`best practices </contributing/community/reviews>`
+to ensure a constructive feedback process.
+
+If you think someone fails to keep this advice in mind and you want another
+perspective, once again please join channel on `Symfony Slack`_. If you
```
Comment (dbrumann) [inline contributing/code/patches.rst]: I think the channel name is missing? 

```suggestion
perspective, once again please join the channel #contrib on `Symfony Slack`_. If you
```

**Comment by wouterj** [comment]
> @wouterj did I handle the BC for the section titles correctly now?

Yes. Except that I gave you wrong instructions (turns out the syntax is `.. _old-title:`, but I'm happy to fix that while merging the PR

**Comment by lsmith77** [comment]
@wouterj hopefully its correct now .. hurray for regexp!

File: contributing/code/patches.rst (hunk 4 above)
Comment (lsmith77) [reply contributing/code/patches.rst]: @dbrumann I reworded the first `must` instead

File: contributing/code/patches.rst (hunk 6 above)
Comment (lsmith77) [reply contributing/code/patches.rst]: thx .. integrated it

**Review by dbrumann — APPROVED** [review]

**Comment by wouterj** [comment]
> @wouterj hopefully its correct now .. hurray for regexp!

Perfect!

File: contributing/code/patches.rst [hunk 7]
```diff
@@ -1,10 +1,21 @@
...
 ==================
 
-Patches are the best way to provide a bug fix or to propose enhancements to
-Symfony.
+A pull request, "PR" for short, is the best way to provide a bug fix or to
+propose enhancements to Symfony.
 
-Step 1: Setup your Environment
+Step 1: Check existing Issues and Pull Requests
+-------------------------------------
```
Comment (javiereguiluz) [inline contributing/code/patches.rst]: This line should be as long as its heading. Otherwise RST format will show an error. Thanks.

Comment (lsmith77) [reply contributing/code/patches.rst]: thx. fixed

**Comment by lsmith77** [comment]
renamed the file and added an entry to the redirection map and hopefully updated all the relative links

**Comment by wouterj** [comment]
Thanks Lukas! I've merged this into 3.4 and will merge it up from there. Also, while merging I solved some more wrong references, hope everything passes now.

**Comment by lsmith77** [comment]
ah .. nevermind .. looks like you already solved the conflict .. so closing

**Pull request closed without merging**